lib/rails/generators/task/templates/task.rb

```diff
@@ -9,10 +9,12 @@
 <% actions.each do |action| -%>
   desc "TODO"
   task <%= action %>: [:environment] do
+    ActiveRecord::Base.transaction do
 
 
-    # DO NOT REMOVE THIS PART. MARKS THE RAKE AS COMPLETE IN THE DATABASE
-    RakeMigration.mark_complete(__FILE__)
+      # DO NOT REMOVE THIS PART. MARKS THE RAKE AS COMPLETE IN THE DATABASE
+      RakeMigration.mark_complete(__FILE__)
```

Comment:
Shouldn't this be outside the transaction block ?

Reply:
@yashBiz Agreed. Any potential breaks in the gem method should not affect the application flow. Made the appropriate changes and merged to develop